src/main/java/org/folio/marc4ld/service/marc2ld/preprocessor/impl/Bib7xxPreprocessor.java

```diff
@@ -0,0 +1,77 @@
+package org.folio.marc4ld.service.marc2ld.preprocessor.impl;
+
+import static org.folio.marc4ld.util.Constants.A;
+import static org.folio.marc4ld.util.Constants.B;
+import static org.folio.marc4ld.util.Constants.S;
+import static org.folio.marc4ld.util.Constants.T;
+import static org.folio.marc4ld.util.Constants.TAG_245;
+import static org.folio.marc4ld.util.Constants.TAG_776;
+import static org.folio.marc4ld.util.MarcUtil.getSubfieldValue;
+import static org.folio.marc4ld.util.MarcUtil.isSubfieldPresent;
+
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.folio.marc4ld.service.marc2ld.preprocessor.DataFieldPreprocessor;
+import org.marc4j.marc.DataField;
+import org.marc4j.marc.MarcFactory;
+import org.marc4j.marc.Record;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class Bib7xxPreprocessor implements DataFieldPreprocessor {
+
+  private static final List<String> TAGS = List.of(TAG_776);
+
+  private final MarcFactory marcFactory;
+
+  @Override
+  public Optional<DataField> preprocess(PreprocessorContext context) {
+    var originalDataField = context.dataField();
+    var preprocessedDataField = duplicateDataField(originalDataField);
+    addSubfieldsRelatedToTitle(context.marcRecord(), preprocessedDataField);
+    return Optional.of(preprocessedDataField);
+  }
+
+  @Override
+  public List<String> getTags() {
+    return TAGS;
+  }
+
+  private DataField duplicateDataField(DataField originalDataField) {
+    var dataField = marcFactory.newDataField(originalDataField.getTag(), originalDataField.getIndicator1(),
+      originalDataField.getIndicator2());
+    originalDataField.getSubfields().forEach(dataField::addSubfield);
+    return dataField;
+  }
+
+  private void addSubfieldsRelatedToTitle(Record marcRecord, DataField dataField) {
+    setSubfieldT(marcRecord, dataField);
+    setSubfieldS(dataField, getSubfieldValue(T, dataField));
+  }
+
+  private void setSubfieldT(Record marcRecord, DataField dataField) {
+    if (isSubfieldPresent(T, dataField)) {
+      return;
+    }
+    var titleFrom245 = getTitle(marcRecord);
+    dataField.addSubfield(marcFactory.newSubfield(T, titleFrom245));
+  }
+
+  private void setSubfieldS(DataField dataField, String subfieldValueT) {
+    if (isSubfieldPresent(S, dataField)) {
+      return;
+    }
+    dataField.addSubfield(marcFactory.newSubfield(S, subfieldValueT));
+  }
+
+  private String getTitle(Record marcRecord) {
+    return marcRecord.getDataFields()
+      .stream()
+      .filter(dataField -> TAG_245.equals(dataField.getTag()))
+      .findFirst()
+      .map(dataField -> getSubfieldValue(A, dataField) + " " + getSubfieldValue(B, dataField))
```

Comment:
Will this line throw exception if 245$b is not present?

Reply:
It will but I thought that it's not a real case? Is possible that 245$a/245$b is not present?

Reply:
Fixed